### PR TITLE
Document what happens when pod uses nonexistent secret.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

File: docs/secrets.md [hunk 1]
```diff
@@ -83,14 +83,24 @@ versions of Kubernetes are expected to provide more automation for populating
...
-Once a pod is created, its secret volumes will not change, even if the secret
-resource is modified.  To change the secret used, the original pod must be
-deleted, and a new pod (perhaps with an identical PodSpec) must be created.
-Therefore, updating a secret follows the same workflow as deploying a new
-container image.  The `kubectl rolling-update` command can be used ([man
-page](kubectl_rolling-update.md)).
+## Secret and Pod Lifetime interaction
+
+When a pod is created via the API, there is no check whether the references
+exists.  It may be scheduled to a node.  Once scheduled, the kubelet will try
```
Comment (goltermann) [inline docs/secrets.md]: "references exist" or "reference exists".

**Comment by erictune** [comment]
fixed.

**Comment by erictune** [comment]
shippable failure seems like a flake.

@pmorie please take a look

File: docs/secrets.md [hunk 2]
```diff
@@ -83,14 +83,24 @@ versions of Kubernetes are expected to provide more automation for populating
...
+
+When a pod is created via the API, there is no check whether a referenced
+secret exists.  Once a pod is scheduled, the kubelet will try to fetch the
+secret value.  If the secret cannot be fetched because it does not exist or
+because of a temporary lack of connection to the API server, kubelet will
+periodically retry.  It will report an event about the pod explaining the
+reason it is not started yet.  Once the a secret is fetched, the kubelet will
+create and mount a volume containing it.  None of the pod's containers will
+start until all the pod's volumes are mounted.
+
```
Comment (lavalamp) [inline docs/secrets.md]: nit: extra blank line

**Comment by lavalamp** [comment]
LGTM

File: docs/secrets.md (hunk 2 above)
Comment (erictune) [inline docs/secrets.md]: ok

**Pull request merged**